### PR TITLE
Improve error reporting and export missing request types in alert sch…

### PR DESCRIPTION
**Pull request by lootag**
…eduler client

 Please enter the commit message for your changes. Lines starting

### CODE REVIEW

File: go/actions-client.go [hunk 1]
```diff
@@ -75,7 +75,11 @@ func (a ActionsClient) Create(ctx context.Context, req *CreateActionRequest) (*a
 	defer conn.Close()
 	client := actions.NewActionsServiceClient(conn)
 
-	return client.CreateAction(callProperties.Ctx, req, callProperties.CallOptions...)
+	response, err := client.CreateAction(callProperties.Ctx, req, callProperties.CallOptions...)
+	if err != nil {
+		return nil, NewSdkAPIError(err, CreateActionRPC, "actions")
```
Comment (celaus) [inline go/actions-client.go]: Please make "actions" a constant in the file

File: go/actions-client.go [hunk 2]
```diff
@@ -103,21 +111,29 @@ func (a ActionsClient) Replace(ctx context.Context, req *ReplaceActionRequest) (
...
+		return nil, NewSdkAPIError(err, ReplaceActionRPC, "actions")
+	}
+	return response, nil
 }
 
 // Delete deletes an action.
 func (a ActionsClient) Delete(ctx context.Context, req *DeleteActionRequest) (*actions.DeleteActionResponse, error) {
 	callProperties, err := a.callPropertiesCreator.GetTeamsLevelCallProperties(ctx)
-	if err != nil {
+	if err == nil {
```
Comment (celaus) [inline go/actions-client.go]: 🤔

File: go/alerts-client.go [hunk 3]
```diff
@@ -684,7 +684,11 @@ func (a AlertsClient) Create(ctx context.Context, req *CreateAlertDefRequest) (*
 	defer conn.Close()
 	client := alerts.NewAlertDefsServiceClient(conn)
 
-	return client.CreateAlertDef(callProperties.Ctx, req, callProperties.CallOptions...)
+	response, err := client.CreateAlertDef(callProperties.Ctx, req, callProperties.CallOptions...)
+	if err != nil {
+		return nil, NewSdkAPIError(err, CreateAlertDefRPC, "alerts")
```
Comment (celaus) [inline go/alerts-client.go]: "alerts" should be a constant

File: go/apikeys-client.go [hunk 4]
```diff
@@ -89,7 +89,11 @@ func (t ApikeysClient) Create(ctx context.Context, req *apikeys.CreateApiKeyRequ
 	defer conn.Close()
 	client := apikeys.NewApiKeysServiceClient(conn)
 
-	return client.CreateApiKey(callProperties.Ctx, req, callProperties.CallOptions...)
+	response, err := client.CreateApiKey(callProperties.Ctx, req, callProperties.CallOptions...)
+	if err != nil {
+		return nil, NewSdkAPIError(err, CreateAPIKeyRPC, "aaa")
```
Comment (celaus) [inline go/apikeys-client.go]: "aaa" should be a constant

File: go/archive-logs.go [hunk 5]
```diff
@@ -64,7 +64,11 @@ func (c ArchiveLogsClient) Update(ctx context.Context, req *archiveLogs.SetTarge
 	defer conn.Close()
 	client := archiveLogs.NewTargetServiceClient(conn)
 
-	return client.SetTarget(callProperties.Ctx, req, callProperties.CallOptions...)
+	response, err := client.SetTarget(callProperties.Ctx, req, callProperties.CallOptions...)
+	if err != nil {
+		return nil, NewSdkAPIError(err, ArchiveLogsSetTargetRPC, "logs")
```
Comment (celaus) [inline go/archive-logs.go]: "logs" should be a constant

File: go/archive-metrics.go [hunk 6]
```diff
@@ -96,7 +96,11 @@ func (c ArchiveMetricsClient) Update(ctx context.Context, req *archiveMetrics.Up
 	defer conn.Close()
 	client := archiveMetrics.NewMetricsConfiguratorPublicServiceClient(conn)
 
-	return client.Update(callProperties.Ctx, req, callProperties.CallOptions...)
+	response, err := client.Update(callProperties.Ctx, req, callProperties.CallOptions...)
+	if err != nil {
+		return nil, NewSdkAPIError(err, ArchiveMetricsUpdateRPC, "metrics")
```
Comment (celaus) [inline go/archive-metrics.go]: "metrics" should be a constant

File: go/archive-retentions.go [hunk 7]
```diff
@@ -62,7 +62,11 @@ func (c ArchiveRetentionsClient) Get(ctx context.Context, req *archiveRetention.
 	defer conn.Close()
 	client := archiveRetention.NewRetentionsServiceClient(conn)
 
-	return client.GetRetentions(callProperties.Ctx, req, callProperties.CallOptions...)
+	response, err := client.GetRetentions(callProperties.Ctx, req, callProperties.CallOptions...)
+	if err != nil {
+		return nil, NewSdkAPIError(err, ArchiveRetentionGetRetentionsRPC, "archive")
```
Comment (celaus) [inline go/archive-retentions.go]: "archive" should be a constant

File: go/dashboard-folders-client.go [hunk 8]
```diff
@@ -63,7 +63,11 @@ func (c DashboardsFoldersClient) Create(ctx context.Context, req *dashboards.Cre
 	defer conn.Close()
 	client := dashboards.NewDashboardFoldersServiceClient(conn)
 
-	return client.CreateDashboardFolder(callProperties.Ctx, req, callProperties.CallOptions...)
+	response, err := client.CreateDashboardFolder(callProperties.Ctx, req, callProperties.CallOptions...)
+	if err != nil {
+		return nil, NewSdkAPIError(err, DashboardFoldersCreateDashboardFolderRPC, "dashboards")
```
Comment (celaus) [inline go/dashboard-folders-client.go]: "dashboards" constant

File: go/dashboards-client.go [hunk 9]
```diff
@@ -1110,7 +1110,11 @@ func (d DashboardsClient) Create(ctx context.Context, req *CreateDashboardReques
 	defer conn.Close()
 	client := dashboards.NewDashboardsServiceClient(conn)
 
-	return client.CreateDashboard(callProperties.Ctx, req, callProperties.CallOptions...)
+	response, err := client.CreateDashboard(callProperties.Ctx, req, callProperties.CallOptions...)
+	if err != nil {
+		return nil, NewSdkAPIError(err, CreateDashboardRPC, "dashboards")
```
Comment (celaus) [inline go/dashboards-client.go]: "dashboards" constant

File: go/data-sets-client.go [hunk 10]
```diff
@@ -70,7 +70,11 @@ func (d DataSetClient) Create(ctx context.Context, req *CreateDataSetRequest) (*
 
 	client := enrichment.NewCustomEnrichmentServiceClient(conn)
 
-	return client.CreateCustomEnrichment(callProperties.Ctx, req, callProperties.CallOptions...)
+	response, err := client.CreateCustomEnrichment(callProperties.Ctx, req, callProperties.CallOptions...)
+	if err != nil {
+		return nil, NewSdkAPIError(err, CreateDataSetRPC, "logs")
```
Comment (celaus) [inline go/data-sets-client.go]: "logs" const

File: go/data-usage-client.go [hunk 11]
```diff
@@ -109,8 +113,11 @@ func (c DataUsageClient) GetLogsCount(ctx context.Context, req *dataUsage.GetLog
 	conn := callProperties.Connection
 	defer conn.Close()
 	client := dataUsage.NewDataUsageServiceClient(conn)
-
-	return client.GetLogsCount(callProperties.Ctx, req, callProperties.CallOptions...)
+	response, err := client.GetLogsCount(callProperties.Ctx, req, callProperties.CallOptions...)
+	if err != nil {
+		return nil, NewSdkAPIError(err, GetLogsCountRPC, "aaa")
```
Comment (celaus) [inline go/data-usage-client.go]: "aaa" should be its own constant in this file (no sharing required)

File: go/dataprime-client.go [hunk 12]
```diff
@@ -41,7 +41,11 @@ func (c DataprimeClient) Query(ctx context.Context, req *QueryRequest) (dataprim
 	defer conn.Close()
 	client := dataprime.NewDataprimeQueryServiceClient(conn)
 
-	return client.Query(callProperties.Ctx, req, callProperties.CallOptions...)
+	response, err := client.Query(ctx, req)
+	if err != nil {
+		return nil, NewSdkAPIError(err, dataprime.DataprimeQueryService_Query_FullMethodName, "dataprime")
```
Comment (celaus) [inline go/dataprime-client.go]: Please re-export the name as DataPrimeQueryRPC and use that. "dataprime" should be a constant too

File: go/enrichments-client.go [hunk 13]
```diff
@@ -91,7 +91,11 @@ func (e EnrichmentsClient) Add(ctx context.Context, req *AddEnrichmentsRequest)
 	defer conn.Close()
 	client := enrichment.NewEnrichmentServiceClient(conn)
 
-	return client.AddEnrichments(callProperties.Ctx, req, callProperties.CallOptions...)
+	response, err := client.AddEnrichments(callProperties.Ctx, req, callProperties.CallOptions...)
+	if err != nil {
+		return nil, NewSdkAPIError(err, AddEnrichmentsRPC, "enrichments")
```
Comment (celaus) [inline go/enrichments-client.go]: "enrichments" const

File: go/events2metrics-client.go [hunk 14]
```diff
@@ -160,7 +160,11 @@ func (e Events2MetricsClient) Create(ctx context.Context, req *CreateE2MRequest)
 	defer conn.Close()
 	client := e2m.NewEvents2MetricServiceClient(conn)
 
-	return client.CreateE2M(callProperties.Ctx, req, callProperties.CallOptions...)
+	response, err := client.CreateE2M(callProperties.Ctx, req, callProperties.CallOptions...)
+	if err != nil {
+		return nil, NewSdkAPIError(err, E2MCreateRPC, "events2metrics")
```
Comment (celaus) [inline go/events2metrics-client.go]: "events2metrics" const

File: go/groups-client.go [hunk 15]
```diff
@@ -117,7 +121,11 @@ func (c GroupsClient) Get(ctx context.Context, req *groups.GetTeamGroupRequest)
 	defer conn.Close()
 	client := groups.NewTeamPermissionsMgmtServiceClient(conn)
 
-	return client.GetTeamGroup(callProperties.Ctx, req, callProperties.CallOptions...)
+	response, err := client.GetTeamGroup(callProperties.Ctx, req, callProperties.CallOptions...)
+	if err != nil {
+		return nil, NewSdkAPIError(err, GetTeamGroupRPC, "aaa")
```
Comment (celaus) [inline go/groups-client.go]: "aaa" should also be a const in this file

File: go/integrations-client.go [hunk 16]
```diff
@@ -131,7 +131,11 @@ func (c IntegrationsClient) Create(ctx context.Context, req *SaveIntegrationRequ
 	defer conn.Close()
 	client := ext.NewIntegrationServiceClient(conn)
 
-	return client.SaveIntegration(callProperties.Ctx, req, callProperties.CallOptions...)
+	response, err := client.SaveIntegration(callProperties.Ctx, req, callProperties.CallOptions...)
+	if err != nil {
+		return nil, NewSdkAPIError(err, SaveIntegrationRPC, "integrations")
```
Comment (celaus) [inline go/integrations-client.go]: "integrations" should be a const too

File: go/notifications-client.go [hunk 17]
```diff
@@ -235,7 +235,11 @@ func (c NotificationsClient) CreateConnector(ctx context.Context, req *CreateCon
 	defer conn.Close()
 	client := connectores.NewConnectorsServiceClient(conn)
 
-	return client.CreateConnector(callProperties.Ctx, req, callProperties.CallOptions...)
+	response, err := client.CreateConnector(callProperties.Ctx, req, callProperties.CallOptions...)
+	if err != nil {
+		return nil, NewSdkAPIError(err, ConnectorsCreateRPC, "notifications")
```
Comment (celaus) [inline go/notifications-client.go]: "notifications" should be a const

File: go/recording-rule-group-client.go [hunk 18]
```diff
@@ -78,7 +78,11 @@ func (r RecordingRuleGroupSetsClient) Create(ctx context.Context, req *CreateRul
 	defer conn.Close()
 	client := recordingRuleGroups.NewRuleGroupSetsClient(conn)
 
-	return client.Create(callProperties.Ctx, req, callProperties.CallOptions...)
+	response, err := client.Create(callProperties.Ctx, req, callProperties.CallOptions...)
+	if err != nil {
+		return nil, NewSdkAPIError(err, CreateRuleGroupSetRPC, "recording-rules")
```
Comment (celaus) [inline go/recording-rule-group-client.go]: "recording-rules" const

File: go/roles-client.go [hunk 19]
```diff
@@ -72,7 +72,11 @@ func (r RolesClient) Create(ctx context.Context, req *CreateRoleRequest) (*roles
 	defer conn.Close()
 	client := roles.NewRoleManagementServiceClient(conn)
 
-	return client.CreateRole(callProperties.Ctx, req, callProperties.CallOptions...)
+	response, err := client.CreateRole(callProperties.Ctx, req, callProperties.CallOptions...)
+	if err != nil {
+		return nil, NewSdkAPIError(err, RolesCreateRoleRPC, "roles")
```
Comment (celaus) [inline go/roles-client.go]: "roles" const

File: go/rules-groups-client.go [hunk 20]
```diff
@@ -211,7 +211,11 @@ func (r RuleGroupsClient) Create(ctx context.Context, req *CreateRuleGroupReques
 	defer conn.Close()
 	client := rulegroups.NewRuleGroupsServiceClient(conn)
 
-	return client.CreateRuleGroup(callProperties.Ctx, req, callProperties.CallOptions...)
+	response, err := client.CreateRuleGroup(callProperties.Ctx, req, callProperties.CallOptions...)
+	if err != nil {
+		return nil, NewSdkAPIError(err, RuleGroupsCreateRuleGroupRPC, "rules")
```
Comment (celaus) [inline go/rules-groups-client.go]: "rules" const

File: go/saml-client.go [hunk 21]
```diff
@@ -66,7 +66,11 @@ func (s SamlClient) GetSPParameters(ctx context.Context, req *saml.GetSPParamete
 	defer conn.Close()
 	client := saml.NewSamlConfigurationServiceClient(conn)
 
-	return client.GetSPParameters(callProperties.Ctx, req, callProperties.CallOptions...)
+	response, err := client.GetSPParameters(callProperties.Ctx, req, callProperties.CallOptions...)
+	if err != nil {
+		return nil, NewSdkAPIError(err, saml.SamlConfigurationService_GetSPParameters_FullMethodName, "aaa")
```
Comment (celaus) [inline go/saml-client.go]: "aaa" another const in here. re-export the SamlConfigurationServiceGetSPParamsRPC

File: go/scopes-client.go [hunk 22]
```diff
@@ -82,7 +82,11 @@ func (c ScopesClient) Create(ctx context.Context, req *CreateScopeRequest) (*sco
 	defer conn.Close()
 	client := scopes.NewScopesServiceClient(conn)
 
-	return client.CreateScope(callProperties.Ctx, req, callProperties.CallOptions...)
+	response, err := client.CreateScope(callProperties.Ctx, req, callProperties.CallOptions...)
+	if err != nil {
+		return nil, NewSdkAPIError(err, CreateScopeRPC, "aaa")
```
Comment (celaus) [inline go/scopes-client.go]: "aaa" another const

File: go/slos-client.go [hunk 23]
```diff
@@ -121,7 +121,11 @@ func (c SLOsClient) Create(ctx context.Context, req *slos.CreateServiceSloReques
 	defer conn.Close()
 	client := slos.NewServiceSloServiceClient(conn)
 
-	return client.CreateServiceSlo(callProperties.Ctx, req, callProperties.CallOptions...)
+	response, err := client.CreateServiceSlo(callProperties.Ctx, req, callProperties.CallOptions...)
+	if err != nil {
+		return nil, NewSdkAPIError(err, SloCreateRPC, "infra-monitoring")
```
Comment (celaus) [inline go/slos-client.go]: "infra-monitoring" const

File: go/tco-policies-client.go [hunk 24]
```diff
@@ -163,7 +163,11 @@ func (t TCOPoliciesClient) Create(ctx context.Context, req *CreatePolicyRequest)
 	defer conn.Close()
 	client := tcopolicies.NewPoliciesServiceClient(conn)
 
-	return client.CreatePolicy(callProperties.Ctx, req, callProperties.CallOptions...)
+	response, err := client.CreatePolicy(callProperties.Ctx, req, callProperties.CallOptions...)
+	if err != nil {
+		return nil, NewSdkAPIError(err, TCOPoliciesCreatePolicyRPC, "tco")
```
Comment (celaus) [inline go/tco-policies-client.go]: "tco" const

File: go/teams-client.go [hunk 25]
```diff
@@ -72,7 +72,11 @@ func (c TeamsClient) Create(ctx context.Context, req *CreateTeamInOrgRequest) (*
 	defer conn.Close()
 	client := teams.NewTeamServiceClient(conn)
 
-	return client.CreateTeamInOrg(callProperties.Ctx, req, callProperties.CallOptions...)
+	response, err := client.CreateTeamInOrg(callProperties.Ctx, req, callProperties.CallOptions...)
+	if err != nil {
+		return nil, NewSdkAPIError(err, CreateTeamInOrgRPC, "aaa")
```
Comment (celaus) [inline go/teams-client.go]: "aaa" const

File: go/webhooks-client.go [hunk 26]
```diff
@@ -266,7 +266,12 @@ func (c WebhooksClient) Create(ctx context.Context, req *CreateOutgoingWebhookRe
 	defer conn.Close()
 	client := webhooks.NewOutgoingWebhooksServiceClient(conn)
 
-	return client.CreateOutgoingWebhook(callProperties.Ctx, req, callProperties.CallOptions...)
+	response, err := client.CreateOutgoingWebhook(callProperties.Ctx, req, callProperties.CallOptions...)
+	if err != nil {
+		return nil, NewSdkAPIError(err, OutgoingWebhookCreateRPC, "integrations")
```
Comment (celaus) [inline go/webhooks-client.go]: "integrations" const

File: rust/cx-sdk/src/client/actions.rs [hunk 27]
```diff
@@ -106,6 +106,7 @@ impl ActionsClient {
             .map_err(|status| SdkApiError {
                 status,
                 endpoint: "/com.coralogixapis.actions.v2.ActionsService/CreateAction".into(),
+                feature_group: "actions".into(),
```
Comment (celaus) [inline rust/cx-sdk/src/client/actions.rs]: a single const would be better in the file

**Review by celaus — COMMENTED** [review]
please make the feature group ids constants in the client files

File: go/actions-client.go (hunk 2 above)
Comment (lootag) [reply go/actions-client.go]: Shoot, I was wondering why a test failed. Thanks, fixed :-)

**Review by celaus — APPROVED** [review]

**Pull request merged**